### ISSUE TITLE
Trace NLD native output trajectories on the reversed-ridge counterexample

### ISSUE THREAD

**Issue by rinsakamo** [rinsakamo/relay-self#127]
## Summary

Measure the observable native output-token trajectories on the strongest explicit counterexample left by #125, without introducing a second inference implementation.

Parent: #101.
Predecessor: #125 (`TOKEN_BUDGET_ISOLATION_PASS`).

## Fixed focus subject

```text
A = FLEE(destination=ridge)
B = FLEE(destination=cave)
C = DEFER

Cave route open: false
Ridge route open: true

expected A -> ridge
max_new_tokens = 32
```

#125 observed:

```text
AR          -> explicit B -> cave
dLM         -> explicit A -> ridge
Linear-Spec -> explicit B -> cave
```

## Method

Reuse the current qualified reversed-mapping token-budget physical runner unchanged as the inference path.

A fresh transaction still records the full 72-call source matrix. Add observation-only fields to each raw output:

- generated token ids;
- per-token decoded text;
- bounded parser state after every generated-token prefix.

Then analyze only the 18 `reversed_ridge_open / max_new_tokens=32` observations.

## Questions

For each native mode record:

- final label/destination/correctness;
- stable output-sequence fingerprints across six repetitions;
- first generated-token prefix carrying any bounded label;
- first explicit decision prefix;
- NFE / latency / token count.

For each mode pair record the first output-token divergence and common prefix length.

## Evidence boundary

This gate observes native output tokens only. It does not expose hidden states, attention, logits, draft internals, or verifier internals.

First textual divergence does not prove an internal causal mechanism.

If output-level trajectory remains insufficient to explain the mode difference, internal forward-path instrumentation may be proposed as a separate later gate.

## Completion

Complete when one fresh transaction records a valid 72-call physical source trace plus a valid 18-observation focus trajectory analysis, or truthfully terminates as `FAIL_NOT_QUALIFIED`.

Expected structural success status:

```text
NATIVE_OUTPUT_TRAJECTORY_PASS
```

Wrong model decisions remain valid evidence.

Refs #101.
Refs #125.
Refs #104.

**Comment by rinsakamo** [comment]
Implementation is merged on current main via PR #129. Current main: `405037ccbbe646640cb329bcb7c3231af83e6f56`; tree: `4de0e90a31ae9e6fea818395655b95750f4ddf5d`; open PRs: 0. The physical transaction now reuses the qualified #125 72-call source runner, adds token-prefix parser observability to raw outputs, and analyzes the 18 `reversed_ridge_open / max_new_tokens=32` observations. Canonical run: `EVIDENCE=/tmp/relay-self-nld-trajectory-$(date -u +%Y%m%dT%H%M%SZ); bash experiments/run_nld_native_path_trajectory_transaction.sh --evidence-root "$EVIDENCE"; cat "$EVIDENCE/summary.json"; cat "$EVIDENCE/trajectory-analysis.json"`. Success status is `NATIVE_OUTPUT_TRAJECTORY_PASS`. Do not close #127 before fresh physical evidence.

**Comment by rinsakamo** [comment]
## Terminal reconciliation — NATIVE_OUTPUT_TRAJECTORY_PASS

Fresh physical evidence completed on current authority.

- HEAD: 405037ccbbe646640cb329bcb7c3231af83e6f56
- tree: 4de0e90a31ae9e6fea818395655b95750f4ddf5d
- branch: main
- working tree: clean; open PRs: 0
- ruleset: 23442682 active; squash-only; required checks repository-contracts, pytest, lint
- PR #129: merged
- venv: /home/rinsa/.venvs/nld
- HF_HOME: /home/rinsa/models/huggingface
- Python: 3.12.3
- torch: 2.14.0+cu130
- Transformers: 5.17.0
- GPU: NVIDIA GeForce RTX 3060, 12,288 MiB
- driver: 591.44
- model: nvidia/Nemotron-Labs-Diffusion-3B
- dtype: bf16; device: CUDA; seed: 1; LoRA off; quantization none; serving framework none
- warm-cache model load: 8.818912338989321 seconds; model loaded once
- source physical: 72/72 measured calls; 6 warm-ups excluded
- focus: 18/18 observations for reversed_ridge_open at max_new_tokens=32
- evidence: /tmp/relay-self-nld-trajectory-20260919T004611Z
- raw source: /tmp/relay-self-nld-trajectory-20260919T004611Z/actual-model.json
- derived analysis: /tmp/relay-self-nld-trajectory-20260919T004611Z/trajectory-analysis.json

Focus subject:
A = FLEE(destination=ridge)
B = FLEE(destination=cave)
C = DEFER
cave open = false; ridge open = true; expected A -> ridge.

Mode summaries below use six observations per mode. Latency is median / nearest-rank p95 in milliseconds. NFE is median / p95.

- AR: count 6; correct 0/6; labels B x6; destinations cave x6; sequence hash 5960809b657142ed4a059a2f810ba684388fc54efdb8af9946b08c0d70d92813 x6; latency 1001.974 / 1042.702 ms; NFE 32 / 32; first any-label token 7; first explicit-decision token 7; parse source explicit x6.
- dLM: count 6; correct 6/6; labels A x6; destinations ridge x6; sequence hash 9d8ea41f07b52c0bfb2275120dbd854c1d06ca64f302be36dd57fb7c7d45b698 x6; latency 937.579 / 968.598 ms; NFE 26 / 26; first any-label token 6; first explicit-decision token 6; parse source explicit x6.
- Linear Self-Speculation: count 6; correct 0/6; labels B x6; destinations cave x6; sequence hash 11ca0434d516591b8726518ed9dfe0495ee162f381efb2018fba8e80389882c8 x6; latency 998.350 / 1019.926 ms; NFE 27 / 27; first any-label token 7; first explicit-decision token 7; parse source explicit x6.

Pairwise observable output trajectories:

- AR vs dLM: common prefix 1 token; first divergence at token 2. AR emitted token 1784, text "The"; dLM emitted token 1438, text "**".
- AR vs Linear Self-Speculation: common prefix 32 tokens. They are identical through the shorter AR output; there is no divergence within the shared output. Linear Self-Speculation continues at token 33 with token 1319, text " (".
- dLM vs Linear Self-Speculation: common prefix 1 token; first divergence at token 2. dLM emitted token 1438, text "**"; Linear Self-Speculation emitted token 1784, text "The".

Representative focus outputs:
- AR: "\nThe decision label is **B**.\n\n**Reasoning:** ..."
- dLM: "\n**Decision label: A**\n\n**Reasoning:** ..."
- Linear Self-Speculation: "\nThe decision label is **B**.\n\n**Reasoning:** ..."

Bounded interpretation:
The observable native trajectories separate from the beginning for dLM versus both AR and Linear Self-Speculation: the first token is shared, and token 2 differs. AR and Linear Self-Speculation are identical through all 32 AR-capped tokens in the focus representative sequence, with Linear Self-Speculation emitting one additional token. Within each mode, all six focus sequences are stable. The first explicit bounded decision appears at token 6 for dLM and token 7 for AR and Linear Self-Speculation, and the final labels remain stable across repetitions.

This is observable token-level evidence only. It does not identify a hidden causal decision point, hidden-state/attention/logit/draft/verifier behavior, product ranking, or adaptive policy. Model output remains distinct from World truth and Action authorization; #104 remains the external consequence-correction boundary.

Status: NATIVE_OUTPUT_TRAJECTORY_PASS. #127 completion condition is satisfied by the fresh 72-call source trace and 18-observation focus analysis.
